R/bootstrap.nlsfit.R

```diff
@@ -379,10 +397,19 @@ bootstrap.nlsfit <- function(fn,
   }else{
     ## the format of gr has to be nrows=length(par), ncols=length(Y)
     if(errormodel == "yerrors"){
+      grpriors <- c()
+      if(is.null(priors$param)==FALSE && is.null(priors$p)==FALSE && is.null(priors$psamples)==FALSE){
```

Comment:
You can write `!x` instead of `x == FALSE`. But are they really `NULL` if there are no priors passed? I'd rather think that you should test for `length(priors$param) > 0`, `length(priors$p) > 0` and `ncol(priors$samples) > 0`.

Actually make a check that `length(priors$param) == length(priors$p) == ncol(priors$samples)` for consistency at the beginning of the function where all the other `stopifnot` are placed. And once you have that check, you just have to check either of the list elements to see whether there are priors.

Reply:
Now the consistency check for the elements of 'priors' is implemented. I guess that checking whether they are NULL is correct, because the elements of 'priors' are set to c() per default (see function bootstrap.nlsfit).

Reply:
Be careful. `is.null(priors$param)` is okay, but `ncol(NULL)` is not the same as `length(NULL)`:

```
> length(NULL)
[1] 0
> ncol(NULL)
NULL
> ncol(as.matrix(NULL))
Fehler in array(x, c(length(x), 1L), if (!is.null(names(x))) list(names(x),  : 
  'data' muss vom Typ vector sein, war 'NULL'
```

Reply:
I just fixed this bug by introducing if/ else statements
```
  if(!is.null(priors$psamples)){
    ncolps <- ncol(as.matrix(priors$psamples))
  }else{
    ncolps <- length(priors$psamples)
  }

  stopifnot( length(priors$param) == length(priors$p) &&
             length(priors$param) == ncolps &&
             length(priors$p) == ncolps )
```
and now the Travis CI checks succeed again.